daemon/policy.go

```diff
@@ -276,7 +277,41 @@ func (d *Daemon) PolicyDelete(labels labels.LabelArray) *apierror.APIError {
 		return apierror.New(DeletePolicyNotFoundCode, "policy not found")
 	}
 
-	d.TriggerPolicyUpdates([]policy.NumericIdentity{})
+	go func() {
+		wg := d.TriggerPolicyUpdates([]policy.NumericIdentity{})
+		// Wait for all policies to be updated so that
+		// we can grab a fresh map of consumables.
+		wg.Wait()
+
+		consumables := d.consumableCache.GetConsumables()
+
+		endpointmanager.Mutex.RLock()
+		for _, ep := range endpointmanager.Endpoints {
+			ep.Mutex.RLock()
+			// If the policy is not being enforced then keep the CT
+			// entries
+			// FIXME: Do we really need to check the daemon.PolicyEnforcement()?
+			if ep.SecLabel == nil ||
+				!ep.Opts.IsEnabled(endpoint.OptionPolicy) ||
+				d.PolicyEnforcement() == endpoint.NeverEnforce {
+				ep.Mutex.RUnlock()
+				continue
+			}
+			epSecID := ep.SecLabel.ID
+
+			ep.Mutex.RUnlock()
+
+			idsToKeep := map[uint32]bool{}
+			if consumers, ok := consumables[epSecID]; ok {
+				for _, consumer := range consumers {
+					idsToKeep[consumer.Uint32()] = true
+				}
+			}
+			log.Debugf("Keeping entries of EP %d: %+v", ep.ID, idsToKeep)
+			endpointmanager.KeepCTEntryOf(!d.conf.IPv4Disabled, ep, idsToKeep)
```

Comment:
A new sec ID can appear while we construct this table and update it? In which case we might miss a new security identity popping up. No? Do we need to hold a lock to avoid updating consumables in the meantime or is it guaranteed that we pick up such changes later?